daemon/cmd/datapath.go

```diff
@@ -401,12 +401,12 @@ func (d *Daemon) initMaps() error {
 
 	ipv4Nat, ipv6Nat := nat.GlobalMaps(option.Config.EnableIPv4,
 		option.Config.EnableIPv6)
-	if option.Config.EnableIPv4 {
+	if ipv4Nat != nil {
 		if _, err := ipv4Nat.Create(); err != nil {
 			return err
 		}
 	}
-	if option.Config.EnableIPv6 {
+	if ipv6Nat != nil {
```

Comment:
[1]

Reply:
Why? `nat.GlobalMaps` check the preconditions to create the maps. No need to duplicate all of them in the caller.

Reply:
Simply to avoid removing the interface.

Reply:
Going back to the original commit that introduced the interface, 319ccde9f19deaa435dd6d29a84ee13cfcea90fa, I don't see any explanation of what it was trying to do with the interface. In general, interfaces help to abstract but looking at the code in this PR, all of the touched packages are already directly depending on bpf so I don't see much difference there.

At some point I'd love for us to come up with a general bpf mock map abstraction, but I don't think this specific `NatMap` interface is achieving that today so unless there's some clear argument why it needs to be here, I'm OK to remove it.

Reply:
> Going back to the original commit that introduced the interface, 319ccde, I don't see any explanation of what it was trying to do with the interface. In general, interfaces help to abstract but looking at the code in this PR, all of the touched packages are already directly depending on bpf so I don't see much difference there.

It might have got copied over from the `nat` package: https://github.com/cilium/cilium/blob/e2d9cbb93f326c7017b0177ed7c5fbb1557f6970/pkg/maps/nat/nat.go#L71-L79

My understanding is that these interfaces are the used in `cilium/cmd/*`:

https://github.com/cilium/cilium/blob/e2d9cbb93f326c7017b0177ed7c5fbb1557f6970/cilium/cmd/bpf_nat_list.go#L49-L59

 so that we can replace the actual maps with the mocks in the tests:
 
 https://github.com/cilium/cilium/blob/e2d9cbb93f326c7017b0177ed7c5fbb1557f6970/cilium/cmd/bpf_nat_list_test.go#L87-L99

That said it looks like the `bpf ct` commands are using the `CtMap` interface:

https://github.com/cilium/cilium/blob/e2d9cbb93f326c7017b0177ed7c5fbb1557f6970/cilium/cmd/bpf_ct_list.go#L133-L141

So I also think it should be fine to get rid of this in the `ctmap` package